Add Cube Inverter, update MC Text Gen and Outline Creator

SirJain0 opened the pull request:
# Cube Inverter
A simple plugin that inverts cubes with a click of the button.

There are many cases in which inverted cubes are useful (besides outlines), but making them is not too convenient as of now. Also, UVs will get messed up, which requires manual fixing. This plugin fixes that.

To use this plugin, press the button that appears in the size fields of the cubes (located next to the inflation box). The action is also keybinded for extra convenience.

# MC Text Generator
**Changelog:**
- Reword some incorrect or vague descriptions for some of the text settings
- Make layer setting only appear when the depth setting is 0
- Got rid of max limit for depth of characters
- Changed default thickness of characters
- Reworded the empty text field warning's description
- Added new characters: `$`, `/`
- Mark plugin as deprecated

# Outline Creator
**Changelog:**
- Added support for meshes

Malik12tree commented:
Basically the same as the [Outline Creator](https://www.blockbench.net/plugins/outline_creator) plugin?

SirJain0 commented:
No.
The outline creator plugin specifically adds another element and makes it slightly bigger which creates that effect.
Mine changes the dimensions of the already-present cube, which you currently have to do manually.

SirJain0 commented:
Just updated this PR to include changes to the Outline Creator and MC Text Generator plugins.
@GitWither Just pinging you in advance - could you please review the changes made to your plugin? The update just adds support for meshes. Thanks.

GitWither commented:
I'm ok with the changes to my plugin

JannisX11 commented:
Please start making individual PRs per plugin. Unless it's just the same change across different plugins. It helps me manage and review them faster.

SirJain0 commented:
Will do for the future.

SirJain0 commented:
(Typo in above commit's name - meant to say `MC Text Generator: Add "/" Character`)

JannisX11 commented on plugins/mc_text_generator.js:
```diff
@@ -24,8 +24,8 @@
         about: "This plugin adds a button under the `Tools` menu that allows you to generate Minecraft-like text.\n## How to use\nTo use this plugin, go to `Tools > Generate Text`. Simply enter some text, configure your settings how you like, and press `Generate`!\n\nPlease report any bugs or suggestions you may have.",
-        tags: ["Minecraft", "Font", "Generator"],
-        version: "2.0.0",
+        tags: ["Deprecated", "Minecraft", "Font"],
```
Tags are not updated in plugins.json

JannisX11 commented on plugins/cube_inverter.js:
```diff
@@ -0,0 +1,174 @@
+                        cube.faces.south.texture = cube.faces.north.texture
+                        cube.faces.north.cullface = cube.faces.south.cullface 
+                        cube.faces.south.cullface = cube.faces.north.cullface
+                        cube.faces.north.rotation += 180
```
You might want to make sure that the rotation stays within the permitted values 0, 90, 180, 270.
Modulo solution: `rotation = (rotation + 180) % 360`

JannisX11 commented on plugins/outline_creator.js:
```diff
@@ -101,7 +97,28 @@ function createOutline(outline_thickness) {
+        mesh.name = mesh.name + "_outline";
+    }
+
+    Canvas.updateAll();
```
Canvas.updateAll is quite inefficient and will be deprecated at some point. I'd recommend to use Canvas.updateView API instead, with all affected elements and the transform and geometry aspects: https://github.com/JannisX11/blockbench/blob/master/js/modeling/transform.js#L676

SirJain0 commented:
@JannisX11 Reviewed and made changes, thanks

SirJain0 commented:
Sorry, I realized I overcomplicated the process of the elements aspect - just fixed that

JannisX11 commented:
The extra button in the element size toolbar concerns me. On default sidebar width it causes the toolbar to overflow, causing the entire layout below to shift. This can shift the outliner around when selecting things. That's not a great default user experience.
Try and add this button to another toolbar or menu instead.

SirJain0 commented:
Good point. I've added the action to the tools tab now, which I think makes the most sense.
Also fixed the missing `about` information in the plugins.json.

SirJain0 commented:
Realized I made an error with the modulo fix for cube UV rotations. Just fixed that.
@JannisX11 I think this should be good, but also let me know if you think another tab would be better for the action for the Cube Inverter plugin.
